src/subscription/subscription.service.ts

```diff
@@ -469,22 +452,63 @@ export class SubscriptionService {
     }
   }
 
-  /**
-   * It creates a new instance of the DevTestRequestDto class, and then saves it to the database
-   * @param {DevTestRequestDto} testData - DevTestRequestDto
-   */
-  async recordTest(testData): Promise<void> {
-    const newTest = this.devTestingRepo.create(testData);
-    this.devTestingRepo.save(newTest);
-  }
+  /* Finding a record in the database that matches the apiId, method, and route. */
+  async saveTest(
+    // apiId: string,
+    profileId: string,
+    body: DevTestRequestDto,
+  ): Promise<DevTesting> {
+    const test = await this.devTestingRepo.findOne({
+      where: {
+        apiId: body.apiId,
+        method: body.method,
+        route: encodeURIComponent(body.route),
+      },
+    });
 
+    if (test) {
```

Comment:
we dont need to check if a similar test alreadt exists, users can add multiple tests for the same route and method